seacrowd/sea_datasets/total_defense_meme/total_defense_meme.py

```diff
@@ -0,0 +1,277 @@
+                | set(annotation["SG_Memes"])
+            )
+        )
+
+        # annotation data is a list of dict, instead of dict of image names
+        def get_value(image_name, list_of_dicts):
+            for dictionary in list_of_dicts:
+                if image_name in dictionary:
+                    return dictionary[image_name]
+            return None
+
+        key = 0
+        for image_name in image_names:
+            # assert image exist in directory
+            assert (image_dir / image_name).exists(), f"Image {image_name} not found"
+            image_path = str(image_dir / image_name)
+
+            # get categories, can be multiple
+            categories = []
+            if image_name in annotation["Non_Memes"]:
+                categories.append("Non_Memes")
+            if image_name in annotation["Non_SG_Memes"]:
+                categories.append("Non_SG_Memes")
+            if image_name in annotation["SG_Memes"]:
+                categories.append("SG_Memes")
+
+            # get attributes
+            text = get_value(image_name, annotation["Text"])
+            tags = get_value(image_name, annotation["Tags"])
+            raw_pillar_stances = get_value(image_name, annotation["Pillar_Stances"])
+
+            # process pillar stances
+            pillar_stances = []
+            if raw_pillar_stances:
+                for pillar, stances in raw_pillar_stances:
+                    category = pillar.split(" ")[0]
+                    pillar_stances.append({"category": category, "stance": stances})
+
+            # source schema
+            if self.config.schema == "source":
+                yield key, {
+                    "image_path": image_path,
+                    "categories": categories,
+                    "text": text,
+                    "tags": tags,
+                    "pillar_stances": pillar_stances,
+                }
+                key += 1
+
+            # ocr seacrowd schema
+            elif self.config.schema == _SEACROWD_SCHEMA["OCR"]:
+                yield key, {
+                    "id": str(key),
+                    "image_paths": [image_path],
+                    "texts": text,
+                    "metadata": {
+                        "tags": tags,
+                        "pillar_stances": pillar_stances,
+                    },
+                }
+                key += 1
+
+            # pillar classification seacrowd schema
+            elif self.config.schema == _SEACROWD_SCHEMA["IMC_MULTI"]:
+                if pillar_stances:  # only those with pillar stances
+                    yield key, {
+                        "id": str(key),
+                        "labels": [pillar["category"] for pillar in pillar_stances],
+                        "image_path": image_path,
+                        "metadata": {
+                            "tags": tags,
+                            "stances": [pillar["stance"] for pillar in pillar_stances],
```

Comment:
Could you please make this `"stances"` --> `"agreed_stances"` and add another variable `"all_stances"`?

The `all_stances` labels have all the stance annotations while the `agreed_stances` labels are the "correct labels" based on [this processing detail in Section 3.3 Quality Control Measures from the paper](https://arxiv.org/pdf/2305.17911.pdf).

```
Lastly, the annotators annotate the meme’s stances towards the assigned pillars: support, against, or neutral. To ensure the reliability of the dataset, each meme is annotated by two annotators. If the disagreements contain similar opinions, the overlap annotations will be considered correct labels. However, if there are disagreements with entirely different perspectives, a third annotator will be brought in to provide an additional annotation for the meme. The overlapping annotations between at least two annotators will then be considered the correct labels. In the extreme case where all three annotators have different opinions, the meme will be flagged and removed from the dataset.
```

Reply:
> In the extreme case where all three annotators have different opinions, the meme will be flagged and removed from the dataset.

Actually it's not. Just for an example, this is the raw data from `annotation.json` (see img_4124).
![image](https://github.com/SEACrowd/seacrowd-datahub/assets/64322471/ab1c903b-0819-4d0a-80b6-f8e802f0e650)

> To ensure the reliability of the dataset, each meme is annotated by two annotators. If the disagreements contain similar opinions, the overlap annotations will be considered correct labels. However, if there are disagreements with entirely different perspectives, a third annotator will be brought in to provide an additional annotation for the meme. The overlapping annotations between at least two annotators will then be considered the correct labels.

There is also no "agreed stances" properties in the dataset. Except if we want to add a script to process the dataset. But IMO, since we are just "loading" the dataset, the processing should be given later to the user.

Reply:
Also to add about the inconsistencies from the dataset, see below.
- img_4129: perfect example for three annotators, we can get obtain the agreed stances.
- img_4130: where is the third annotator?
- img_4131: first and second annotators already agreed on neutral, why third annotator?
![image](https://github.com/SEACrowd/seacrowd-datahub/assets/64322471/16927aa6-5f51-4937-88a9-719c5dcca552)